pkg/components/validate_headers.go

```diff
@@ -24,16 +25,21 @@ func contains(s []string, target string) bool {
 	return false
 }
 
-func incomingMatchesAllowed(allowed map[string][]string, incoming map[string][]string) error {
+func incomingMatchesAllowed(allowed map[string][]string, incoming map[string][]string, split map[string]string) error {
 	checkedHeaderAndValues := make(map[string][]string)
-	for allowedKey, allowedValues := range allowed {
+	for allowedHeader, allowedValues := range allowed {
 		// check if incoming header values have a configured allowed header present and search through them if so
-		if matchedIncomingHeaderValues, present := incoming[http.CanonicalHeaderKey(allowedKey)]; present {
+		if matchedIncomingHeaderValues, present := incoming[http.CanonicalHeaderKey(allowedHeader)]; present {
+			delimiter := " "
```

Comment:
Question: I'm not happy about this part, specifically setting a default of " ". What I really want to do is set this default to empty string, "". However, if i do this, then within the contains function, we split on "", which results in ->
"split" -> "s","p","l","i","t". This is the best i could come up with. Does anyone have any suggestions? :)

Reply:
Yes, I would assume if no delimiter is explicitly set, then the entire value should be used, without splitting.

Suggestion: update the code such that the allowedValues are all checked against the entire string when no explicit delimiter is specified. 

Of course, this means the only combination that makes sense or that would ever work when no delimiter is given is a single allowed value, but that seems acceptable to me. 🤷‍♂

Reply:
Agree with @faldridge.